include/setup.h

```diff
@@ -304,23 +304,16 @@ class Prop_hex final : public Property {
 
 typedef void (*SectionFunction)(Section*);
 
+struct SectionFunctions {
+	SectionFunction init = nullptr;
+	SectionFunction destroy = nullptr;
+	bool execute_on_config_change = false;
```

Comment:
It's not an enum.  It's using all of these fields.

Reply:
I see GitHub is a bit confusing here, since I only selected line 310 for comment, not the whole struct. I meant just replacing the "bool"-type with an enum-type. 

Point is that if I see something like "{DOS_init, DOS_destroy, true}", I have no clue what "true" means until I look at the definition of the struct. If I see for example "{DOS_init, DOS_destroy, OnConfigChange::execute}", I have a clue what the intention is. And we have the opportunity to easily add more options if needed.

Reply:
@japsmits That's a good suggestion—for the future.

The config handling will be extensively refactored/rewritten. Until then, we only fix critical bugs in it, otherwise we're just spinning our wheels making small incremental improvements to something that will be switched out eventually anyway.

Until then, we use this "named flag" approach to avoid in passing in nondescript "true" and "false" literals.

But yeah, ultimately, we'll go with the approach you described in the rewrite. Enums all the way 😎